packages/exchange-core/src/test/Matching.test.ts

```diff
@@ -0,0 +1,551 @@
+import { IRECDeviceService } from '@energyweb/utils-general';
+import { assert } from 'chai';
+import { List } from 'immutable';
+
+import { Matching } from '../Matching';
+import { Order, OrderSide } from '../Order';
+import { Trade } from '../Trade';
+import { Product } from '../Product';
+import { Ask } from '../Ask';
+import { Bid } from '../Bid';
+
+interface IOrderCreationArgs {
+    product?: Product;
+    price?: number;
+    volume?: number;
+}
+
+interface ITestCase {
+    bidsBefore: Order[];
+    asksBefore: Order[];
+
+    expectedTrades: Trade[];
+
+    asksAfter?: Order[];
+    bidsAfter?: Order[];
+}
+
+describe('Matching tests', () => {
+    const deviceService = new IRECDeviceService();
+
+    const twoUSD = 2;
+    const onekWh = 1000;
+    const deviceVintage = 2019;
+    const locationCentral = ['Thailand;Central;Nakhon Pathom'];
+    const locationEast = ['Thailand;East;Nakhon Pathom'];
+
+    const solarTypeLevel1 = deviceService.encode([['Solar']]);
+    const solarTypeLevel2 = deviceService.encode([['Solar', 'Photovoltaic']]);
+    const solarTypeLevel3 = deviceService.encode([['Solar', 'Photovoltaic', 'Classic silicon']]);
+    const solarTypeLevel32 = deviceService.encode([['Solar', 'Photovoltaic', 'Roof mounted']]);
+
+    const windTypeLevel1 = deviceService.encode([['Wind']]);
+    const windTypeLevel2 = deviceService.encode([['Wind', 'Onshore']]);
+    const windTypeLevel22 = deviceService.encode([['Wind', 'Offshore']]);
+
+    let initialOrderId = 0;
+
+    const createAsk = (args?: IOrderCreationArgs) => {
+        return new Ask(
+            (initialOrderId++).toString(),
+            args?.price || twoUSD,
```

Comment:
I think it's better to use `??` instead of `||` here, because if someone wants to specify 0 price which is a valid number, it will fall back to `twoUSD` currently, because of `||`.